avaframe/com1DFA/DFAfunctionsCython.pyx

```diff
@@ -218,29 +211,30 @@ def computeForceC(cfg, particles, fields, dem, dT, int frictType):
       nx, ny, nz = getVector(Lx0, Ly0, w[0], w[1], w[2], w[3], nxArray, nyArray, nzArray)
       nx, ny, nz = normalize(nx, ny, nz)
 
-      # add artificial viscosity
-      vMeanx, vMeany, vMeanz = getVector(Lx0, Ly0, w[0], w[1], w[2], w[3], VX, VY, VZ)
-      # compute normal component of the velocity
-      vMeanNorm = scalProd(vMeanx, vMeany, vMeanz, nx, ny, nz)
-      # remove normal component (make sure vMean is in the tangent plane)
-      vMeanx = vMeanx - vMeanNorm * nx
-      vMeany = vMeany - vMeanNorm * ny
-      vMeanz = vMeanz - vMeanNorm * nz
-      # compute particle to field velocity difference
-      dvX = vMeanx - ux
-      dvY = vMeany - uy
-      dvZ = vMeanz - uz
-      dvMag = norm(dvX, dvY, dvZ)
-      Alat = 2.0 * math.sqrt((m * h) / rho)
-      fDrag = (subgridMixingFactor * 0.5 * rho * dvMag * Alat * dt) / m
-
-      # update velocity with artificial viscosity - implicit method
-      ux = ux + fDrag * vMeanx
-      uy = uy + fDrag * vMeany
-      uz = uz + fDrag * vMeanz
-      ux = ux / (1.0 + fDrag)
-      uy = uy / (1.0 + fDrag)
-      uz = uz / (1.0 + fDrag)
+      if viscOption == 1:
+        # add artificial viscosity
+        vMeanx, vMeany, vMeanz = getVector(Lx0, Ly0, w[0], w[1], w[2], w[3], VX, VY, VZ)
+        # compute normal component of the velocity
+        vMeanNorm = scalProd(vMeanx, vMeany, vMeanz, nx, ny, nz)
+        # remove normal component (make sure vMean is in the tangent plane)
+        vMeanx = vMeanx - vMeanNorm * nx
+        vMeany = vMeany - vMeanNorm * ny
+        vMeanz = vMeanz - vMeanNorm * nz
+        # compute particle to field velocity difference
+        dvX = vMeanx - ux
+        dvY = vMeany - uy
+        dvZ = vMeanz - uz
+        dvMag = norm(dvX, dvY, dvZ)
+        Alat = 2.0 * math.sqrt((m * h) / rho)
+        fDrag = (subgridMixingFactor * 0.5 * rho * dvMag * Alat * dt) / m
+
+        # update velocity with artificial viscosity - implicit method
+        ux = ux + fDrag * vMeanx
+        uy = uy + fDrag * vMeany
+        uz = uz + fDrag * vMeanz
+        ux = ux / (1.0 + fDrag)
+        uy = uy / (1.0 + fDrag)
+        uz = uz / (1.0 + fDrag)
```

Comment:
this is @matthiasto fucntion. I do prefer updqte this in q next PR or let him do

Reply:
We can make this a separate function but we can not put both viscosity methods in the same place simply because the Ata viscosity is intrinsec to the sph gradient computation and is added in the sph function